include/CoolPropTools.h

```diff
@@ -341,54 +341,62 @@
         Dictionary(){};
-        bool is_empty(void){return numbers.empty() && strings.empty() && double_vectors.empty() && string_vectors.empty();}
-        void add_string(std::string s1, std::string s2){ strings.insert(std::pair<std::string, std::string>(s1, s2));}
-        void add_number(std::string s1, double d){ numbers.insert(std::pair<std::string, double>(s1, d));}
-        void add_double_vector(std::string s1, std::vector<double> d){ double_vectors.insert(std::pair<std::string, std::vector<double> >(s1, d));}
-        void add_string_vector(std::string s1, std::vector<std::string> d){ string_vectors.insert(std::pair<std::string, std::vector<std::string> >(s1, d));}
-        std::string get_string(std::string s)
+        bool is_empty(void) const {return numbers.empty() && strings.empty() && double_vectors.empty() && string_vectors.empty();}
+        void add_string(const std::string &s1, const std::string &s2){ strings.insert(std::pair<std::string, std::string>(s1, s2));}
+        void add_number(const std::string &s1, double d){ numbers.insert(std::pair<std::string, double>(s1, d));}
+        void add_double_vector(const std::string &s1, const std::vector<double> &d){ double_vectors.insert(std::pair<std::string, std::vector<double> >(s1, d));}
+        void add_string_vector(const std::string &s1, const std::vector<std::string> &d){ string_vectors.insert(std::pair<std::string, std::vector<std::string> >(s1, d));}
+        std::string get_string(const std::string &s) const
         {
-            if (strings.find(s) != strings.end()){ 
-                return strings[s]; 
-            } 
-            else{ 
-                throw CoolProp::ValueError(format("%s could not be matched in get_string",s.c_str())); 
+            strings_map::const_iterator i = strings.find(s);
+            if (i != strings.end()){
+                return i->second;
+            }
+            else{
+                throw CoolProp::ValueError(format("%s could not be matched in get_string",s.c_str()));
             }
         };
-        double get_double(std::string s)
+        double get_double(const std::string &s) const
         {
-            if (numbers.find(s) != numbers.end()){ 
-                return numbers[s]; 
-            } 
-            else{ 
-                throw CoolProp::ValueError(format("%s could not be matched in get_number",s.c_str())); 
+            numbers_map::const_iterator i = numbers.find(s);
+            if (i != numbers.end()){
+                return i->second;
+            }
+            else{
+                throw CoolProp::ValueError(format("%s could not be matched in get_number",s.c_str()));
             }
         };
-        double get_number(std::string s)
+        double get_number(const std::string &s) const
         {
             return get_double(s);
         };
-        const std::vector<double>& get_double_vector(std::string s)
+        const std::vector<double>& get_double_vector(const std::string &s) const
```

Comment:
the rightmost const makes this function "constant function": it means that
1. we declare that the function does not change the state of the object.
2. thus, compiler does not allow us to change any of object's members, and only allows to call their "const" functions;

and only this allows to call this function, when the object itself is const.